Спринт 12. Рейтинг Фильмов. Добавление базы данных.

SergikF opened the pull request:
Добавление базы данных к проекту java-filmorate .
Рефакторинг проекта для обеспечения работы с базой данных.

SergikF committed:
Спринт 12-1. Рейтинг Фильмов.
Промежуточное задание 12 спринта.

SergikF commented:
проверка добавления комментария ...

SergikF commented:
Предварительная, промежуточная задача.
Разработка структуры базы данных. Загружена инфограмма.

Just-Roman commented:
Проверка

Just-Roman commented:
1. По требованию 2NF - Таблица должна иметь ключ, это относится ко всем таблицам кроме users, films, rating_mpa.
2. Название таблиц правильнее писать в единственном числе.

SergikF commented:
> 1. По требованию 2NF - Таблица должна иметь ключ, это относится ко всем таблицам кроме users, films, rating_mpa.
> 2. Название таблиц правильнее писать в единственном числе.

Спасибо. Рассмотрю данное замечание.

SergikF committed:
Спринт 12-1. Рейтинг Фильмов.
Промежуточное задание 12 спринта.

SergikF committed:
Спринт 12-1. Рейтинг Фильмов.
Промежуточное задание 12 спринта.

SergikF commented:
> 1. По требованию 2NF - Таблица должна иметь ключ, это относится ко всем таблицам кроме users, films, rating_mpa.
_Согласен, что таблица по правилам должна иметь первичный ключ, для уникальности каждой записи, но с моей точки зрения необязательно для этого заводить отдельное поле. Для этих целей я буду использовать составной первичный ключ._
![image](https://github.com/user-attachments/assets/d81acdc2-f488-4f12-8af6-22dd43c901d5)

> 2. Название таблиц правильнее писать в единственном числе.
_Данный вопрос строго не регламентируется и остаётся свободным, как вопрос стиля. Единственное требование - однообразие для всей базы данных. Поэтому изменил названия таблиц исходя из требований однообразия стиля._
![er_diagram_filmrate](https://github.com/user-attachments/assets/55b93f43-ab29-4001-b3bf-370d3d0bcc35)

Just-Roman commented:
Супер!

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка стиля по GitHub.

Alexander-Bolotov commented:
> > 1. По требованию 2NF - Таблица должна иметь ключ, это относится ко всем таблицам кроме users, films, rating_mpa.
> >    _Согласен, что таблица по правилам должна иметь первичный ключ, для уникальности каждой записи, но с моей точки зрения необязательно для этого заводить отдельное поле. Для этих целей я буду использовать составной первичный ключ._
> >    ![image](https://private-user-images.githubusercontent.com/116564864/409099338-d81acdc2-f488-4f12-8af6-22dd43c901d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk4NTY1MTksIm5iZiI6MTczOTg1NjIxOSwicGF0aCI6Ii8xMTY1NjQ4NjQvNDA5MDk5MzM4LWQ4MWFjZGMyLWY0ODgtNGYxMi04YWY2LTIyZGQ0M2M5MDFkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxOFQwNTIzMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MGJmMDJlNTI5MTIxMTVkM2ZmOTM5YjQxNTg0NjQ5NDBkZGIyMWQ5NmQ3YTFhNDEwMzg2N2RhNzIxYjllYjNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.iMkrlV-TcEn51qnGDtFCymLEItIKUJo_TlszmG84rpE)
> 
> > 2. Название таблиц правильнее писать в единственном числе.
> >    _Данный вопрос строго не регламентируется и остаётся свободным, как вопрос стиля. Единственное требование - однообразие для всей базы данных. Поэтому изменил названия таблиц исходя из требований однообразия стиля._
> >    ![er_diagram_filmrate](https://private-user-images.githubusercontent.com/116564864/409099516-55b93f43-ab29-4001-b3bf-370d3d0bcc35.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk4NTY1MTksIm5iZiI6MTczOTg1NjIxOSwicGF0aCI6Ii8xMTY1NjQ4NjQvNDA5MDk5NTE2LTU1YjkzZjQzLWFiMjktNDAwMS1iM2JmLTM3MGQzZDBiY2MzNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxOFQwNTIzMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02Yzg4MDliNzcxODJhMDJmYjE0ZWNiMmNjNzI5NjBkMTg1MjM3NDk2MTAyNDdmNzYyYjVhMmM3NzQ1NDkzNTM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.86R6bDHU41K3Le2-IdW3Z1R5zt80wZXZTLJXOTBq8I8)

да для friends вполне подходит составной ключ!

Alexander-Bolotov commented on src/main/java/ru/yandex/practicum/filmorate/controller/GenreController.java:
```diff
@@ -0,0 +1,29 @@
+public class GenreController {
+    private final GenreService genreService;
+
+    // получаем список жанров
```
На мой взгляд такие комментарии излишние, код должен быть написан так что бы они не потребовались)
либо нужно использовать формат javadoc

Alexander-Bolotov commented on src/main/java/ru/yandex/practicum/filmorate/controller/GenreController.java:
```diff
@@ -0,0 +1,29 @@
+
+import java.util.List;
+
+@Slf4j
```
нигде в классе не используется логирование

Alexander-Bolotov commented on src/main/java/ru/yandex/practicum/filmorate/service/GenreService.java:
```diff
@@ -0,0 +1,24 @@
+
+import java.util.List;
+
+@Slf4j
```
тоже если не используется, то нет смысла ее ставить

Alexander-Bolotov commented on src/main/java/ru/yandex/practicum/filmorate/storage/GenreDbStorage.java:
```diff
@@ -0,0 +1,50 @@
+        try {
+            return jdbc.query("SELECT * FROM genres", genreRowMapper);
+        } catch (EmptyResultDataAccessException e) {
+            throw new NotFoundException("Список жанров пуст");
```
обычно принято, если запрашивается весь список, то возвращать пустой LIst в таком случае, а исключение выбрасывать, если запрашивается, что то конкретное

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка по замечаниям ревьювера.

SergikF committed:
Спринт 12. Рейтинг Фильмов.
Перевод проекта на хранение данных в базе данных H2.
Корректировка по замечаниям ревьювера.
